examples/gzip/verify.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -e
+
+export NAME=gzip
+
+PWD="$(dirname "${BASH_SOURCE[0]}")"
+
+# shellcheck source=examples/verify-common.sh
+. "${PWD}/../verify-common.sh"
+
+run_log "Test service: localhost:8001/stats/prometheus without compress"
+responds_without_header \
+    "content-encoding: gzip" \
+    http://localhost:8001/stats/prometheus \
+
+run_log "Test service: localhost:8001/stats/prometheus with compress"
+responds_without_header \
+    "content-encoding: gzip" \
+    http://localhost:8001/stats/prometheus \
+    --compressed
+
+run_log "Test service: localhost:8002/stats/prometheus without compress"
+responds_without_header \
+    "content-encoding: gzip" \
+    http://localhost:8002/stats/prometheus \
+
+run_log "Test service: localhost:8002/stats/prometheus with compress"
+set -x
+responds_with_header \
+    "content-encoding: gzip" \
+    http://localhost:8002/stats/prometheus \
+    --compressed
+
+#(TODO daixiang0) 2>&1 can not be parsed well in responds_without, call curl directly here
+run_log "Test service: localhost:8089/plain with compress"
+    curl -s -v --compressed http://localhost:8089/plain 2>&1 | \
+      grep -q "content-encoding: gzip" | [[ "$(wc -l)" -eq 0 ]] || \
+      {
+        echo "ERROR: curl not expect (${*}): $expected" >&2
+        return 1
+      }
+
+run_log "Test service: localhost:8089/json with compress"
+    curl -s -v --compressed http://localhost:8089/json 2>&1 | \
```

Comment:
use `responds_with/out_header`  for these please

Reply:
I tried but fail, since I need response with verb mode which write to stderr, the function can not handle `2>&1`
.

Reply:
i can check again when i next review - we should be able to use them i think - not sure if we cant right now

Reply:
Please help me, I can not use existed function with `2>&1`

Reply:
i can take a look

one thing tho - i would rather we use ` curl -H "Accept-Encoding: gzip"` as that is less curl-specific and better demonstrates what is happening

Reply:
This header do not help.

Reply:
the example needs to not be curl specific - if the example is working correctly then it will work with this header

Reply:
i just tried replacing `--compressed` with the accept header and it works fine

Reply:
Of course, it is a way to verify, is it ok that I use `curl -H "Accept-Encoding: gzip"` directly?

Reply:
use `responds_with/out_*` you can add arguments to curl with it